MapboxNavigation/RouteMapViewController.swift

```diff
@@ -580,6 +583,7 @@ extension RouteMapViewController {
 extension RouteMapViewController: NavigationViewDelegate {
     // MARK: NavigationViewDelegate
     func navigationView(_ view: NavigationView, didTapCancelButton: CancelButton) {
+        self.dismissed = true
```

Comment:
@hactar Should we also set `dismissed` to `true` in deinit as well?

Reply:
Interesting point @vincethecoder, thanks. I don't see a normal scenario that would lead to deinit being called without self.dismissed being set to true already.

That said, there is no harm in setting it to true there too, and that might protect against scenarios where people are doing weird things with the RouteMapViewController.

I'll update the PR accordingly, tough if I understand @akitchen 's comment correctly this code will be obsolete in future anyway.